### PR TITLE
Updated Upstream (Bukkit/CraftBukkit/Spigot)

### PR DESCRIPTION
**Pull request by Machine-Maker**
Upstream has released updates that appear to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
146a7e4b SPIGOT-5345: Add automatic library support

CraftBukkit Changes:
b1064c69 Remove sisu annotation processor from jar
32e40866 SPIGOT-6189: Persistent data disappears when calling setFacingDirection on an item frame
d189f78b #&#x2060;827: Trigger vanilla dimension advancements in non-main worlds
5bbb4a65 Add plumbing for automatic library support

Spigot Changes:
9fb885e8 Rebuild patches

### CODE REVIEW

**Comment by Machine-Maker** [comment]
Someone else def needs to check this over well, not 100% sure I got the loader stuff correct.

**Review by Proximyst — APPROVED** [review]
Looks fine to me. Would like @PaperMC/development-team to also do a proper read through to make sure I didn't miss anything.

**Comment by electronicboy** [comment]
I had a look outside of my ide and it looked fine to me, did wanna have a
peek at the prioritisation patch just to double check but opted to lie down
instead

File: Spigot-Server-Patches/0348-Fix-sounds-when-item-frames-are-modified-MC-123450.patch [hunk 1]
```diff
@@ -18,16 +18,3 @@ index 584f64946821092afab57b9409bc249403bc16e7..43152a6c70c9433d627a58051101530d
...
-diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftItemFrame.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftItemFrame.java
-index 86bb26cd1d327056a64926ed6fc53142ad3756c2..cbdbb89ec28f9b6c3a0eb31be94c440254c6e266 100644
---- a/src/main/java/org/bukkit/craftbukkit/entity/CraftItemFrame.java
-+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftItemFrame.java
-@@ -50,7 +50,7 @@ public class CraftItemFrame extends CraftHanging implements ItemFrame {
-         old.die();
- 
-         EntityItemFrame frame = new EntityItemFrame(world, position, direction);
--        frame.setItem(item);
-+        frame.setItem(item, true, false); // Paper - fix itemframe sound
```
Comment (MiniDigger) [inline Spigot-Server-Patches/0348-Fix-sounds-when-item-frames-are-modified-MC-123450.patch:30]: just to double check, this was dropped intentionally, right?

Comment (Proximyst) [reply Spigot-Server-Patches/0348-Fix-sounds-when-item-frames-are-modified-MC-123450.patch:30]: Believe that was fixed upstream:
https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/32e4086604f6df169869172175a194c41422de38#src/main/java/org/bukkit/craftbukkit/entity/CraftItemFrame.java

**Review by electronicboy — APPROVED** [review]

**Pull request merged**